src/jit/flowgraph.cpp

```diff
@@ -22858,8 +22858,9 @@ void Compiler::fgInsertInlineeBlocks(InlineInfo* pInlineInfo)
     BasicBlock*  iciBlock = pInlineInfo->iciBlock;
     BasicBlock*  block;
 
-    // We can write better assert here. For example, we can check that
-    // iciBlock contains iciStmt, which in turn contains iciCall.
+#ifdef DEBUG
+    noway_assert(iciBlock->Contains(iciStmt) && (iciStmt->gtStmtExpr == iciCall));
```

Comment:
It is a linear check, is not it too expensive here? 
@AndyAyersMS do you remember why it was not added in the first place?

Reply:
>  that's why originally I put it under DEBUG to make sure we won't hit a bad case in RELEASE builds.

Please keep it enabled only for DEBUG/CHECKED build. I mean return `ifdef DEBUG` around `bool Contains(const GenTree* node)` and change `noway_assert` to `assert`.